.github/workflows/train_workflow.yml

```diff
@@ -5,17 +5,35 @@ on:
       script_content:
         description: 'Content of train.py'
         required: true
-        type: string  # Explicitly specify the type
+        type: string
 
 jobs:
   train:
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Cache PyTorch
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/pip
+            ${{ runner.tool_cache }}/Python
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
       - name: Install dependencies
         run: |
-          pip install numpy
-          # pip install torch - need to find a way to cache this otherwise it will take a long time to install
-
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
```

Comment:
we don't need most these requirements to run the file, we mostly need them to run the bot which is not yet properly tested in CI